msglibrary/src/main/java/ingsw/group1/msglibrary/parsing/SMSParsingManager.java

```diff
@@ -0,0 +1,212 @@
+package ingsw.group1.msglibrary.parsing;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import ingsw.group1.msglibrary.SMSMessage;
+
+/**
+ * Class implementing {@link MessageParser} for {@link SMSMessage}.
+ * The class behaves as an Object Pool, requiring a String name to get an appropriate instance.
+ * By default also acts as a parser that does no operation on the messages. Allows setting a new
+ * parser to override the one in use by the current instance by calling
+ * {@link SMSParsingManager#setOverridingParser(Class)}, such a change persists, since it is
+ * stored in {@link android.content.SharedPreferences}, in a file called
+ * {@link SMSParsingManager#PREF_FILENAME} with a key which is:
+ * {@link SMSParsingManager#PREF_KEY_PREFIX} + {@link SMSParsingManager#instanceName}.
+ * <p>
+ * In order to be instantiated the Parser class needs a public constructor with no parameters.
+ * Since the raw class name is stored, generics are not supported. If you need one of the
+ * previously stated features, move the parsing logic to an upper level.
+ * <p>
+ * Also allows resetting to the default parsing strategy by calling
+ * {@link SMSParsingManager#resetOverridingParser()}.
+ *
+ * @author Riccardo De Zen
+ */
+public class SMSParsingManager implements SMSParser {
+
+    //Base name for Preference file name.
+    static final String PREF_FILENAME = "ingsw.group1.msglibrary.parsing.SMSParsingManager";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/RicDeZen/ingsw-group1-message-library/pullRequest?prid=4846460)